polaris/ocean/convergence/spherical/analysis.py

```diff
@@ -0,0 +1,332 @@
+import datetime
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from polaris import Step
+from polaris.ocean.resolution import resolution_to_subdir
+
+
+class SphericalConvergenceAnalysis(Step):
+    """
+    A step for analyzing the output from the geostrophic convergence test
+
+    Attributes
+    ----------
+    resolutions : list of float
+        The resolutions of the meshes that have been run
+
+    icosahedral : bool
+        Whether to use icosahedral, as opposed to less regular, JIGSAW
+        meshes
+
+    dependencies_dict : dict of dict of polaris.Steps
+        The dependencies of this step
+    """
+    def __init__(self, component, resolutions, icosahedral, subdir,
+                 dependencies, convergence_vars):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        component : polaris.Component
+            The component the step belongs to
+
+        resolutions : list of float
+            The resolutions of the meshes that have been run
+
+        icosahedral : bool
+            Whether to use icosahedral, as opposed to less regular, JIGSAW
+            meshes
+
+        subdir : str
+            The subdirectory that the step resides in
+
+        dependencies : dict of dict of polaris.Steps
+            The dependencies of this step
```

Comment:
Here and in the `Attributes` of the class, I think we're going to have to document this in a little more detail now that this is in the framework.  The dependencies must be be keys `mesh`, `init`, and `forward`, each of which is a `dict` with keys that are the resolutions and values that are `Step` objects.  In addition:
* the steps in `mesh` must have an output `base_mesh.nc` that provides the horizontal mesh variables
* the steps in `init` must have an output `initial_state.nc` that provides the initial condition and vertical mesh variables
* the steps in `forward` must have an output `output.nc` that provides the comparison variables at the time of convergence evaluation